dask_expr/tests/test_collection.py

```diff
@@ -94,6 +94,11 @@ def test_dask(pdf, df):
 def test_reductions(func, pdf, df):
     assert_eq(func(df), func(pdf))
     assert_eq(func(df.x), func(pdf.x))
+    # re-conversion to Series is necessary since Dask does not reduce. Trying to
+    # figure out whether this was intended or not
+    # check_dtype False because sub-selection of columns that is pushed through
+    # is not reflected in the meta calculation
+    assert_eq(func(df)["x"], pd.Series(func(pdf)["x"]), check_dtype=False)
```

Comment:
Yeah the main problem we are facing here is:

```
df = pd.DataFrame({"a": [11, 12, 31], "b": 1.0})
ddf = from_pandas(df, npartitions=1)
ddf.a.max() # reduce
ddf.max()["a"] # does not
```

This is what happens in Dask. Since we are transforming the second one into the first one we lose these information. I think Dask should reduce in the second case as well. I'll open an issue in Dask. Wdyt?

Reply:
>I'll open an issue in Dask. Wdyt?

Yes, I think this makes sense as a dask issue.

For the case of dask-expr, `df.a.max()` does reduce, and `df.max()["a"]` should too (but doesn't). As far as I can tell, the problem with the second case is that we are calling `compute` on a dask-expr `Series`, but the underlying expression is being optimized into a `Scalar` collection within that `compute` call (and so the `__dask_postcompute__` logic is converting the scalar back to a `Series`)... Still thinking about the proper "fix" for this.

Reply:
Yeah exactly. The cast-back happens in ``_concat``. One option would be to not cast back anymore if we have len(args) == 1, but this breaks other things. So will have to take a closer look how we can do this efficiently.

Reply:
Looks like we may be able to change `FrameBase.__dask_postcompute__` to the following?:

```python
    def __dask_postcompute__(self):
        state = self.simplify()
        if type(self) != type(state):
            return state.__dask_postcompute__()
        return _concat, ()
```

Wdyt? Does this help with the test?

Reply:
We don't need a new class, that was crab.

Adjusted the meta creation in Projection, which does the trick. Wdyt?

Are we ok with using a pandas utility to check for scalars?